src/backend/ffmpeg/mod.rs

```diff
@@ -0,0 +1,241 @@
+    ) -> Result<Self> {
+        Ok(PseudoBeamerRuleGen { cfg, range })
+    }
+
+    fn finish(
+        self, gen: &mut Generator<'a, SlidesFfmpegEspeak, impl Write>,
+        _peek: Option<WithRange<&Event<'a>>>,
+    ) -> Result<()> {
+        let PseudoBeamerRuleGen { cfg, range } = self;
+        let (diagnostics, backend, mut out) = gen.backend_and_out();
+        let events: Vec<_> = backend.slides.close_until(2, &mut out, range, diagnostics)?;
+        backend.current_frame.extend(events);
+        let events: Vec<_> = backend.slides.open_until(2, cfg, &mut out, range, diagnostics)?;
+        backend.current_frame.extend(events);
+        Ok(())
+    }
+}
+
+#[derive(Debug)]
+pub struct PseudoBeamerHeaderGen<'a> {
+    cfg: &'a Config,
+    level: i32,
+    label: WithRange<Cow<'a, str>>,
+    range: SourceRange,
+}
+
+impl<'a> StatefulCodeGenUnit<'a, SlidesFfmpegEspeak, Header<'a>> for PseudoBeamerHeaderGen<'a> {
+    fn new(
+        cfg: &'a Config, header: WithRange<Header<'a>>,
+        gen: &mut Generator<'a, SlidesFfmpegEspeak, impl Write>,
+    ) -> Result<Self> {
+        let (diagnostics, backend, mut out) = gen.backend_and_out();
+        let WithRange(Header { label, level }, range) = header;
+
+        // close old slide / beamerboxesrounded
+        let events: Vec<_> = backend.slides.close_until(level, &mut out, range, diagnostics)?;
+        backend.current_frame.extend(events);
+
+        write!(out, "\\{}section{{", "sub".repeat(level as usize - 1))?;
+
+        Ok(PseudoBeamerHeaderGen { cfg, level, label, range })
+    }
+
+    fn finish(
+        self, gen: &mut Generator<'a, SlidesFfmpegEspeak, impl Write>,
+        _peek: Option<WithRange<&Event<'a>>>,
+    ) -> Result<()> {
+        let PseudoBeamerHeaderGen { cfg, level, label, range } = self;
+        let (diagnostics, backend, mut out) = gen.backend_and_out();
+        writeln!(out, "}}\\label{{{}}}\n", label.0)?;
+
+        let events: Vec<_> = backend.slides.open_until(level, cfg, &mut out, range, diagnostics)?;
+        backend.current_frame.extend(events);
+        Ok(())
+    }
+}
+
+#[derive(Debug)]
+pub enum CodeBlockGen {
+    Speech(File),
+    Normal(<Beamer as Backend<'static>>::CodeBlock),
+}
+
+impl<'a> StatefulCodeGenUnit<'a, SlidesFfmpegEspeak, CodeBlock<'a>> for CodeBlockGen {
+    fn new(
+        cfg: &'a Config, code_block: WithRange<CodeBlock<'a>>,
+        gen: &mut Generator<'a, SlidesFfmpegEspeak, impl Write>,
+    ) -> Result<Self> {
+        let WithRange(CodeBlock { label: _, caption: _, language }, _range) = &code_block;
+
+        if let Some(WithRange(language, _range)) = language {
+            if language.as_ref() == "espeak" {
+                let (diagnostics, backend, _) = gen.backend_and_out();
+                return backend.open_speech_file(cfg, diagnostics).map(CodeBlockGen::Speech);
+            }
+        }
+
+        <<Beamer as Backend<'static>>::CodeBlock as CodeGenUnit<'a, CodeBlock<'a>>>
```

Comment:
1. No need to specify specific lifetimes here, you can replace both `'static` and `'a` with `'_`.
2. The compiler's inference goes even further, you can just write `CodeGenUnit::new` directly here.

Reply:
Hm, weird. I remember that inference had failed somewhere. But I'll remove those superficial annotations then.